### ISSUE TITLE
Mishandling of wildcard characters in read_parquet

### ISSUE THREAD

**Issue by Malkard** [awslabs/aws-data-wrangler#564]
**Describe the bug**
When trying to read a single file containing wildcard characters (in our case, '['), the read_parquet function fails to find the file:

`wr.s3.read_parquet(path="s3://bucket/some_file[something].parquet")`

results in a NoFilesFound exception

**To Reproduce**
Version 2.4.0
Probably exists since the introduction of wildcards with https://github.com/awslabs/aws-data-wrangler/issues/322

To reproduce: Put a parquet file containing square brackets in a bucket and try to read it.

The implementation should accept a preceding '\' to indicate the _literalness_ of a special character.

**Workaround**
Replace wildcards with '?' before passing to read_parquet.

**Comment by maxispeicher** [comment]
As a quickfix you could just use `glob.escape()` before passing the path:
```python
import glob
import awswrangler as wr

s3_path = "s3://bucket/some_file[something].parquet"
escaped_s3_path = glob.escape(s3_path)
wr.s3.read_parquet(path=escaped_s3_path)
```

**Comment by Malkard** [comment]
Thanks, this works too, and is somewhat cleaner.  

Since the intent of the "path" argument of read_parquet is to handle wildcards, this could be considered as the proper way to deal with this use case, instead of my suggestion.  In which case a simple mention in the documentation might be valuable.

**Comment by igorborgest** [comment]
Released on version [2.5.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/2.5.0)!

**Comment by jtlz2** [comment]
This still doesn't work for me (REF: https://stackoverflow.com/questions/75614048/how-do-i-merge-several-parquet-files-into-one-using-awswrangler) - what am I doing wrong? Thanks!
